internal/fs/type_cache_test.go

```diff
@@ -0,0 +1,390 @@
+// Copyright 2024 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// A collection of tests for a file system where we do not attempt to write to
+// the file system at all. Rather we set up contents in a GCS bucket out of
+// band, wait for them to be available, and then read them via the file system.
+
+package fs_test
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path"
+	"time"
+
+	"github.com/googlecloudplatform/gcsfuse/internal/config"
+	"github.com/googlecloudplatform/gcsfuse/internal/storage/storageutil"
+
+	"github.com/jacobsa/oglematchers"
+	. "github.com/jacobsa/ogletest"
+)
+
+////////////////////////////////////////////////////////////////////////
+// Common
+////////////////////////////////////////////////////////////////////////
+
+type typeCacheTestCommon struct {
+	fsTest
+}
+
+var (
+	// The following should be configured for different tests
+	// differently inside SetUpTestSuite as these need to
+	// set for mount itself.
+
+	// ttlInSeconds is equivalent of metadata-cache:ttl-secs in config-file.
+	ttlInSeconds int64
```

Comment:
Instead of keeping these values global, can we keep inside typeCacheTestCommon ? Every time specific test's SetpTestSuite method will override these values.

Reply:
I tried keeping them in the typeCacheTestCommon class at first. But I'm not sure why it didn't work (values kept changing on their own).  It's the same reason there are so many global vars here too: https://github.com/GoogleCloudPlatform/gcsfuse/blob/342c39b1863fedeb3bd947aaad2d0188dd0737fa/internal/fs/fs_test.go#L97